driver-pulsar/src/main/java/io/nosqlbench/driver/pulsar/ops/PulsarConsumerOp.java

```diff
@@ -136,22 +140,32 @@ public void run(Runnable timeTracker) {
                 }
 
                 // keep track of message ordering and message loss
-                if (seqTracking) {
-                    String msgSeqIdStr = message.getProperties().get(PulsarActivityUtil.MSG_SEQUENCE_ID);
+                String msgSeqIdStr = message.getProperties().get(PulsarActivityUtil.MSG_SEQUENCE_ID);
+                if ( (seqTracking) && !StringUtils.isBlank(msgSeqIdStr) ) {
                     curMsgSeqId = Long.parseLong(msgSeqIdStr);
 
-                    // normal case: message sequence id is monotonically increasing by 1
-                    if ((curMsgSeqId - prevMsgSeqid) == 1) {
-                        prevMsgSeqid = curMsgSeqId;
-                    }
-                    else {
-                        // abnormal case: out of ordering
-                        if (curMsgSeqId < prevMsgSeqid) {
-                            throw new RuntimeException("Detected message ordering is not guaranteed. Older messages are received earlier!");
-                        }
-                        // abnormal case: message loss
-                        else if ( (curMsgSeqId - prevMsgSeqid) > 1 ) {
-                            throw new RuntimeException("Detected message sequence id gap. Some published messages are not received!");
+                    if ( prevMsgSeqId > -1) {
+                        // normal case: message sequence id is monotonically increasing by 1
+                        if ((curMsgSeqId - prevMsgSeqId) != 1) {
+                            // abnormal case: out of ordering
+                            if (curMsgSeqId < prevMsgSeqId) {
+                                throw new RuntimeException("" +
```

Comment:
For each reason to throw an exception, can you create a specific exception type (that extends RuntimeException) and add it to the pulsar driver docs? This allows for configurable error handling.

Reply:
Sure, I will add that

Reply:
Added via commit 949790b797d2d24f22c9f02e7c1a7f013718357c